### PR TITLE
Refactor/clean data fetching

### PR DESCRIPTION
**Pull request by RaphaelEscrig**
This PR refactor all [data fetching](https://nextjs.org/docs/basic-features/data-fetching/overview) for all pages.

Because we have a few data and they are all mocked, no cache revalidation has been set. In a near future, when we will connect the portal to an api, a cache validation is needed.

Feel free to discuss about this implementation !

### CODE REVIEW

**Comment by ErikssonJoakim** [comment]
Am I correct to assume that we will need to change the `getStaticProps` for `getServerSideProps` once the api is up and running? @RaphaelEscrig

**Comment by RaphaelEscrig** [comment]
Not necessarily; `getStaticProps` can be used to fetch external data too. `getServerSideProps` is useful if we want to render a page with data fetched at request time.

With `getStaticProps`, on build time, the page will be cached which is not the default case with `getServerSideProps`.

[Here](https://webkul.com/blog/getstaticprops-vs-getserversideprops-next-js-data-fetching/) is an article explaining the differences between these two methods.

File: environment.d.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-    NAVIGATION_MENU_INTERACT_URL: string
-    NAVIGATION_MENU_LEARN_URL: string
-    NAVIGATION_MENU_OKP4_URL: string
+    NEXT_PUBLIC_NAVIGATION_MENU_HOME_URL: string
+    NEXT_PUBLIC_NAVIGATION_MENU_CREATE_URL: string
+    NEXT_PUBLIC_NAVIGATION_MENU_EXPLORE_URL: string
+    NEXT_PUBLIC_NAVIGATION_MENU_INTERACT_URL: string
+    NEXT_PUBLIC_NAVIGATION_MENU_LEARN_URL: string
+    NEXT_PUBLIC_NAVIGATION_MENU_OKP4_URL: string
     OKP4_DISCORD_URL: string
```
Comment (ErikssonJoakim) [inline environment.d.ts:9]: The social media urls + website do not work, they need to be made public like the navigation variables. I have not tested the other variables

**Review by ErikssonJoakim — CHANGES_REQUESTED** [review]

**Comment by ErikssonJoakim** [comment]
Seeing as we've reworked the api I suggest we close this PR @fredericvilcot

**Pull request closed without merging**